receiver/kafkametricsreceiver/consumer_scraper.go

```diff
@@ -36,13 +37,28 @@ type consumerScraper struct {
 	groupFilter  *regexp.Regexp
 	topicFilter  *regexp.Regexp
 	clusterAdmin sarama.ClusterAdmin
+	saramaConfig *sarama.Config
 	config       Config
 }
 
 func (s *consumerScraper) Name() string {
 	return consumersScraperName
 }
 
+func (s *consumerScraper) start(context.Context, component.Host) error {
+	client, err := newSaramaClient(s.config.Brokers, s.saramaConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create client while starting consumer scraper: %w", err)
+	}
+	clusterAdmin, err := newClusterAdmin(s.config.Brokers, s.saramaConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create cluster admin while starting consumer scraper: %w", err)
```

Comment:
If this returns an error shutdown won't be called. Do we need to call s.client.Close() on the error case? One way to do this would be to do:

```go
defer func(){
   if err != nil && client != nil {
      client.Close()
   }
}()
```

Would have to make err a named return.

Reply:
ha yes, the client should be closed if shutdown won't be called. Will fix now